plugins/processors/k8sdecorator/stores/utils.go

```diff
@@ -30,11 +31,12 @@ func createContainerKeyFromMetric(tags map[string]string) string {
 	return k8sutil.CreateContainerKey(namespace, podName, containerName)
 }
 
-const (
-	// kubeAllowedStringAlphaNums holds the characters allowed in replicaset names from as parent deployment
+var (
+	// deploymentUnallowedRegExp holds the characters allowed in replicaset names from as parent deployment
 	// https://github.com/kubernetes/apimachinery/blob/master/pkg/util/rand/rand.go#L83
-	kubeAllowedStringAlphaNums = "bcdfghjklmnpqrstvwxz2456789"
-	cronJobAllowedString       = "0123456789"
+	deploymentUnallowedRegExp = regexp.MustCompile(`[^b-hj-np-tv-xz-z2-24-9]+`)
+	// cronJobUnallowedRegexp ensures the characters in cron job name are only numbers.
```

Comment:
This is what makes the variable names so confusing. The comment above the regex indicates that the regex is used to "ensure the characters [...] are **only numbers**", but the regular expression itself uses the `^` character to negate that. The statement about the regular expression contradicts (1) the name of the variable, and (2) the actual check the regex performs.

Looking at what is provided by Go's regexp library, there isn't a high level method for matching the regex against the entire string, but that's not necessary because we can construct a regular expression that means the same thing. We can use anchors in the regular expression to denote where the start and end of the string is, for the matching we want. In this case, we want to match `^\d+$`, so that from the beginning to the end of the string, there are only digit characters, and there's at least 1. 

I wrote some quick benchmark tests, too. 
```go
var cronJobUnallowedRegexp = regexp.MustCompile(`[^0-9]+`)
var anchored = regexp.MustCompile(`^\d+$`)

func BenchmarkDoNotMatchAnchored(b *testing.B) {
	s := "123abc" // should NOT match because of "abc"
	for i := 0; i < b.N; i++ {
		anchored.MatchString(s)
	}
	assert.False(b, anchored.MatchString(s))
}

func BenchmarkMatchAnchored(b *testing.B) {
	s := "123456" // should match because only digit chars
	for i := 0; i < b.N; i++ {
		anchored.MatchString(s)
	}
	assert.True(b, anchored.MatchString(s))
}

func BenchmarkMatchRegex(b *testing.B) {
	s := "123abc"
	for i := 0; i < b.N; i++ {
		cronJobUnallowedRegexp.MatchString(s)
	}
	assert.True(b, cronJobUnallowedRegexp.MatchString(s))
}

func BenchmarkDoNotMatchRegex(b *testing.B) {
	s := "123456"
	for i := 0; i < b.N; i++ {
		cronJobUnallowedRegexp.MatchString(s)
	}
	assert.False(b, cronJobUnallowedRegexp.MatchString(s))
}
```

```
goos: darwin
goarch: amd64
pkg: LoadTest
cpu: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
BenchmarkDoNotMatchAnchored
BenchmarkDoNotMatchAnchored-12    	12695895	        89.58 ns/op
BenchmarkMatchAnchored
BenchmarkMatchAnchored-12         	 8242377	       138.4 ns/op
BenchmarkMatchRegex
BenchmarkMatchRegex-12            	 8494279	       142.9 ns/op
BenchmarkDoNotMatchRegex
BenchmarkDoNotMatchRegex-12       	 7746750	       155.3 ns/op
PASS
```

Reply:
Interesting. You are right about anchor though. I need to gather my thoughts a little bit

Reply:
Nitpick - the comments should either drop the variable name reference entirely, or update them to match the new names of the variables.

Reply:
True. Forget to change it after the commit thought. Nice catch and thanks for pointing it out.